src/hooks/hookProxy.ts

```diff
@@ -1,79 +1,496 @@
+        const genericNameMatch = genericName.toUpperCase() === searchDrugName.toUpperCase();
+        
+        if (brandNameMatch || genericNameMatch) {
+          console.log(`Found matching drug: ${brandName} / ${genericName}`);
+          
+          // Look for REMS API endpoint in subjectOf section
+          if (subject.manufacturedProduct[0].subjectOf) {
+            for (const subjectOf of subject.manufacturedProduct[0].subjectOf) {
+              if (subjectOf.document && 
+                  subjectOf.document[0] && 
+                  subjectOf.document[0].title && 
+                  subjectOf.document[0].title[0] && 
+                  subjectOf.document[0].title[0].includes('REMS API') && 
+                  subjectOf.document[0].text && 
+                  subjectOf.document[0].text[0] && 
+                  subjectOf.document[0].text[0].reference) {
+                
+                const referenceValue = subjectOf.document[0].text[0].reference[0];
+                let remsUrl;
+                
+                // Handle different XML formats
+                if (typeof referenceValue === 'string') {
+                  // Direct string format
+                  remsUrl = referenceValue;
+                } else if (referenceValue.$ && referenceValue.$.value) {
+                  // Object with value attribute
+                  remsUrl = referenceValue.$.value;
+                }
+                
+                if (remsUrl) {
+                  // Extract the URL, handling different formats
+                  let cdsServiceUrl;
+                  if (remsUrl.includes(':rems_discovery:')) {
+                    cdsServiceUrl = remsUrl.split(':rems_discovery:')[1];
+                  } else {
+                    cdsServiceUrl = remsUrl;
+                  }
+                  
+                  // Create the return object with drug info and endpoints
+                  const drugInfo = {
+                    brandName: brandName,
+                    genericName: genericName,
+                    remsEndpoint: cdsServiceUrl,                  };
+                  
+                  console.log(`Found REMS endpoint for ${brandName}: ${cdsServiceUrl}`);
+                  return drugInfo;
+                }
+              }
+            }
+          }
+          
+          // Found the drug but no REMS endpoint
+          console.log(`Found drug ${brandName} but no REMS endpoint in XML`);
+          return {
+            brandName: brandName,
+            genericName: genericName,
+            remsEndpoint: null,
+            etasuEndpoint: null
+          };
+        }
+      }
+    }
+    
+    console.log(`Drug ${searchDrugName} not found in XML`);
+    return null;
+  } catch (error) {
+    console.error('Error extracting drug info from XML:', error);
+    return null;
+  }
+}
+
+async function getDrugXmlFromSplZip(rems_spl_date?: string): Promise<any> {
```

Comment:
We should refactor and break apart these huge functions.

Reply:
lets make this follow on tasking for the next release in the interest of time since its more about maintainability than functionality. Will link JIRA ticket once created

Reply:
https://jira.mitre.org/secure/RapidBoard.jspa?rapidView=4984&projectKey=REMS&view=planning&selectedIssue=REMS-814&issueLimit=100#